packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/routing/layout/layout.component.ts

```diff
@@ -0,0 +1,111 @@
+/* eslint-disable no-shadow, no-console */
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import {
+  RouteData,
+  Field,
+  LayoutServiceContextData,
+  getContentStylesheetLink,
+} from '@sitecore-jss/sitecore-jss-angular';
+import { ActivatedRoute } from '@angular/router';
+import { Subscription } from 'rxjs';
+import { JssState } from '../../JssState';
+import { JssMetaService } from '../../jss-meta.service';
+import { JssLinkService } from '../../jss-link.service';
+
+enum LayoutState {
+  Layout,
+  NotFound,
+  Error,
+}
+
+interface RouteFields {
+  [name: string]: unknown;
+  pageTitle: Field<string>;
+}
+
+@Component({
+  selector: 'app-layout',
+  templateUrl: './layout.component.html',
+})
+export class LayoutComponent implements OnInit, OnDestroy {
+  route: RouteData<RouteFields>;
+  state: LayoutState;
+  LayoutState = LayoutState;
+  subscription: Subscription;
+  errorContextData: LayoutServiceContextData;
+  mainClassPageEditing: string;
+
+  constructor(
+    private activatedRoute: ActivatedRoute,
+    private readonly meta: JssMetaService,
+    private linkService: JssLinkService
+  ) {}
+
+  ngOnInit() {
+    // route data is populated by the JssRouteResolver
+    this.subscription = this.activatedRoute.data.subscribe(
+      (data: { jssState: JssState<RouteFields> }) => {
+        if (!data.jssState) {
+          this.state = LayoutState.NotFound;
+          return;
+        }
+
+        if (data.jssState.sitecore && data.jssState.sitecore.route) {
+          this.route = data.jssState.sitecore.route;
+          this.setMetadata(this.route.fields);
+          this.state = LayoutState.Layout;
+          this.mainClassPageEditing = data.jssState.sitecore.context.pageEditing
+            ? 'editing-mode'
+            : 'prod-mode';
+
+          /**  TODO: get contextId and edgeUrl properly **/
+          const sitecoreEdgeContextId = '';
+          const sitecoreEdgeUrl = '';
+          const contentStyles = getContentStylesheetLink(
+            data.jssState.sitecore,
```

Comment:
Can you double-check this type? The first parm is a `LayoutServiceData` type, so think it may need to be:
```suggestion
            { sitecore: data.jssState.sitecore },
```

Reply:
Yes you're right.